### PR TITLE
vec: get a metric with labels that are arbitrarily ordered

### PR DESCRIPTION
**Pull request by overvenus**
Get a metric with labels that are arbitrarily ordered.

/cc @syepes

### CODE REVIEW

**Comment by siddontang** [comment]
what is this PR for?

**Comment by overvenus** [comment]
These is a bug in `get_meric_with()`.
If we create a MetricsVec with unsorted labels, like `&["b", "a"]`,
then `get_metric_with(&labels!{"b" => "2", "a" => "1",})`,
it will return a metric that labels are `b => 1, a=> 2`, because `&labels!{"b" => "2", "a" => "1",}` are wrongly sorted to `&["1", "2"]` in `get_meric_with()`.

**Comment by siddontang** [comment]
Please add test.

File: src/vec.rs [hunk 1]
```diff
@@ -130,19 +129,27 @@ impl<T: MetricVecBuilder> MetricVecCore<T> {
...
-                }
+                None => return Err(Error::Msg(format!("label name {} missing in label map", name))),
             }
-
         }
 
         Ok(h.finish())
     }
 
+    fn sort_value<'a>(&self, labels: &'a HashMap<&str, &str>) -> Result<Vec<&'a str>> {
```
Comment (siddontang) [inline src/vec.rs]: it is not for sorting value, should think a better name.

**Comment by syepes** [comment]
@overvenus Thanks for the fix, it looks like it has resolved my issue.

**Comment by siddontang** [comment]
LGTM
PTAL @disksing

**Comment by disksing** [comment]
LGTM

**Pull request merged**